indexing-service/src/main/java/org/apache/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -1648,6 +1649,30 @@ public int getTotalCapacity()
     return getWorkers().stream().mapToInt(workerInfo -> workerInfo.getWorker().getCapacity()).sum();
   }
 
+  @Override
+  public int getMaximumCapacity()
```

Comment:
Maybe add a javadoc here listing the cases when this method returns -1.

Reply:
I wonder if this method body should be commoned out between `RemoteTaskRunner` and `HttpTaskRunner`.

Reply:
were you thinking of introducing a new class to do this?